inc/bmp/BMP.h

```diff
@@ -0,0 +1,289 @@
+     * Assignable macros :
+     * - BMP5_ENABLE
+     * - BMP5_DISABLE
+     */
+    uint8_t set_fifo_iir_t;
+
+    /*! Fifo pressure IIR
+     * Assignable macros :
+     * - BMP5_ENABLE
+     * - BMP5_DISABLE
+     */
+    uint8_t set_fifo_iir_p;
+};
+struct bmp5_sensor_data
+{
+    /*! Pressure data */
+    float pressure;
+
+    /*! Temperature data */
+    float temperature;
+};
+struct bmp5_iir_config
+{
+    /*! Temperature IIR
+     * Assignable macros :
+     * - BMP5_IIR_FILTER_BYPASS
+     * - BMP5_IIR_FILTER_0_20980
+     * - BMP5_IIR_FILTER_0_08045
+     * - BMP5_IIR_FILTER_0_03695
+     * - BMP5_IIR_FILTER_0_01785
+     * - BMP5_IIR_FILTER_0_00880
+     * - BMP5_IIR_FILTER_0_00435
+     * - BMP5_IIR_FILTER_0_00220
+     */
+    uint8_t set_iir_t;
+
+    /*! Pressure IIR
+     * Assignable macros :
+     * - BMP5_IIR_FILTER_BYPASS
+     * - BMP5_IIR_FILTER_0_20980
+     * - BMP5_IIR_FILTER_0_08045
+     * - BMP5_IIR_FILTER_0_03695
+     * - BMP5_IIR_FILTER_0_01785
+     * - BMP5_IIR_FILTER_0_00880
+     * - BMP5_IIR_FILTER_0_00435
+     * - BMP5_IIR_FILTER_0_00220
+     */
+    uint8_t set_iir_p;
+
+    /*! Shadow IIR selection for temperature
+     * Assignable macros :
+     * - BMP5_ENABLE
+     * - BMP5_DISABLE
+     */
+    uint8_t shdw_set_iir_t;
+
+    /*! Shadow IIR selection for pressure
+     * Assignable macros :
+     * - BMP5_ENABLE
+     * - BMP5_DISABLE
+     */
+    uint8_t shdw_set_iir_p;
+
+    /*! IIR flush in forced mode enable
+     * Assignable macros :
+     * - BMP5_ENABLE
+     * - BMP5_DISABLE
+     */
+    uint8_t iir_flush_forced_en;
+};
+
+
+int8_t bmp5_set_power_mode(enum bmp5_powermode powermode, struct bmp5_dev *dev);
+int8_t bmp5_set_osr_odr_press_config(const struct bmp5_osr_odr_press_config *osr_odr_press_cfg, struct bmp5_dev *dev);
+int8_t bmp5_get_interrupt_status(uint8_t *int_status, struct bmp5_dev *dev);
+int8_t bmp5_set_regs(uint8_t reg_addr, const uint8_t *data, uint32_t len, struct bmp5_dev *dev);
+int8_t bmp5_soft_reset(struct bmp5_dev *dev);
+int8_t bmp5_init(struct bmp5_dev *dev);
+int8_t bmp5_get_regs(uint8_t reg_addr, uint8_t *data, uint32_t len, struct bmp5_dev *dev);
+int8_t bmp5_get_power_mode(enum bmp5_powermode *powermode, struct bmp5_dev *dev);
+int8_t bmp5_get_iir_config(struct bmp5_iir_config *iir_cfg, struct bmp5_dev *dev);
+int8_t bmp5_get_sensor_data(struct bmp5_sensor_data *sensor_data,
+                            const struct bmp5_osr_odr_press_config *osr_odr_press_cfg,
+                            struct bmp5_dev *dev);
+int8_t bmp5_get_osr_odr_press_config(struct bmp5_osr_odr_press_config *osr_odr_press_cfg, struct bmp5_dev *dev);
+
+
+class BMP581
+{
+    public:
+        // Constructor
+        BMP581();
+
+        // Sensor initialization, must specify communication interface
+        //int8_t beginI2C(uint8_t address = BMP5_I2C_ADDR_SEC, TwoWire& wirePort = Wire);
+        int8_t beginI2C(uint8_t address = BMP5_I2C_ADDR_SEC, WIRE_TEENSY& wirePort = Wire_r);
```

Comment:
Changer à I2C_INTERFACE.
Retirer le channel par défaut

```suggestion
        int8_t beginI2C(uint8_t address = BMP5_I2C_ADDR_SEC, I2C_INTERFACE* wirePort);
```